### PR TITLE
Add timeout feature for RPC calls, revert channel check on produce()

### PR DESCRIPTION
**Pull request by PuKoren**
Fixes #49 #48

### CODE REVIEW

File: test/producer-consumer-spec.js [hunk 1]
```diff
@@ -78,77 +78,46 @@ describe('producer/consumer', function () {
...
-      return consumer.consume(fixtures.routingKey, function (message) {
-          assert.equal(message.content, 'ok');
-        })
-        .then(() => {
-          return producer.produce(fixtures.routingKey, {content: 'ok'}, {routingKey: 'route'});
+  describe('rpc timeouts', function() {
+    it('should reject on timeout, if no answer received', function() {
+      return producer.produce('non-existing-queue', { msg: 'ok' }, { rpc: true, timeout: 1000 })
+        .catch((e) => {
+          assert(e);
```
Comment (mrister) [inline test/producer-consumer-spec.js]: nitbit: Maybe check exactly for Timeout Error in this test.

**Comment by mrister** [comment]
Tests failing are related to mocha updates (no more using done callback and returning promises). Once fixed every test is passing.

**Comment by jkernech** [comment]
Aside from the tests failures, it looks good. 
Waiting for the tests to pass to approve the PR.

**Comment by mrister** [comment]
LGTM

**Comment by PuKoren** [comment]
Tests updated, bump for review @jkernech @ky23

**Comment by mrister** [comment]
LGTM

**Comment by jkernech** [comment]
👍 LGTM !

**Comment by ykhatal** [comment]
LGTM

**Pull request merged**